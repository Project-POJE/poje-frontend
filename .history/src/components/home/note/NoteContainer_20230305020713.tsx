export default function NoteContainer() {
  return <div></div>;
}
