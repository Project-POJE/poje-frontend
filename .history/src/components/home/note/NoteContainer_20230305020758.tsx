import styled from 'styled-components';

export default function NoteContainer() {
  return <div></div>;
}

const Container = styled.section``;
