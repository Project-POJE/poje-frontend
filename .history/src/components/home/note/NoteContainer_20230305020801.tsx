import styled from 'styled-components';

export default function NoteContainer() {
  return <Container></Container>;
}

const Container = styled.section``;
