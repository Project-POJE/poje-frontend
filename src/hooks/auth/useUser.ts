import { accessTokenState } from '@src/store/auth/auth';
import { useRecoilValue, useRecoilState, useResetRecoilState } from 'recoil';

export default function useUser() {
  const [user, setUser] = useRecoilState(accessTokenState);
  const resetUser = useResetRecoilState(accessTokenState);
  const clearUser = () => {
    resetUser();
  };

  return { user, clearUser, setUser };
}
