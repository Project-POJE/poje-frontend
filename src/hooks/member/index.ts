import service from '@src/service';
import { useQuery } from '@tanstack/react-query';

export const usePortfolioListsByLike = (page: string) => {
  const { data } = useQuery(
    ['portfoliosListWithLike', page],
    () => service.job.getePortfolioCardsWithLike({ page }),
    { staleTime: 1000 * 60, suspense: true }
  );

  return { pageingUtil: data!.paging, pfAndMemberResp: data!.pfAndMemberResp };
};
