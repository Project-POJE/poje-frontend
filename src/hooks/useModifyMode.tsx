import { useState } from 'react';

export default function useModifyMode() {
  const [isModifyMode, setIsModifyMode] = useState<boolean>(false);
  const toggleModify = () => {
    setIsModifyMode((e) => !e);
  };
  return { isModifyMode, toggleModify };
}
