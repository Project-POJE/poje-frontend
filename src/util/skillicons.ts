import { SkillItemType } from '@src/service/types/portfolio';

export type SkillIconSetType = {
  name: string;
  path: string;
};

export type SkillIconType = 'frontend' | 'backend';

export type IconListType = Record<SkillIconType, SkillItemType[]>;

export type IconListKeyType = keyof IconListType;

type SkillTypeAndNameType = {
  type: IconListKeyType;
  name: string;
  path: string;
};

export function extractSkillIconFromFolder() {
  //skill_icon 하위 폴더들의 이미지들을 가져옴
  const imgContext = require.context(
    '/public/public_assets/skill_icon',
    true,
    /.png$|.ipg$/
  );

  const icons: IconListType = {
    frontend: [],
    backend: [],
  };

  //./backend/node.png 이런애들이 Keys()에 있음
  imgContext.keys().forEach((k) => {
    const extractedSkill = extractSkillTypeAndName(k);
    console.log(k);
    if (extractedSkill) {
      const name = extractedSkill.name;
      const path = extractedSkill.path;
      icons[extractedSkill.type].push({ name, path });
    }
  });
  console.log(icons);

  return icons;
}

// 동적으로 키에 배열을 작성하고 난 후 그 키를 동적으로 가져오는 것은 불가능
function extractSkillTypeAndName(path: string): SkillTypeAndNameType | null {
  // 직업식별자에 포함된 이미지의 식별자를 가져오기
  let skillType, skillName, skillPath;
  let result = path.replace('./', ''); //추출된 스킬 front/css.png;
  const splitPathList = result.split('/');

  if (splitPathList.length === 1) return null;
  skillType = splitPathList[0] as IconListKeyType;
  skillName = splitPathList[1].split('/')[0].split('.')[0];
  skillPath = splitPathList[1].split('/')[0];
  return { type: skillType, name: skillName, path: skillPath };
}
