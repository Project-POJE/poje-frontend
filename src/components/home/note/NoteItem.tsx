import { NoteType } from '@src/service/types/member';
import styled from 'styled-components';

export default function NoteItem(data: NoteType) {
  const { message, sender, sendStatus, sendTime } = data;
  return (
    <Note>
      <Header>
        <Status status={sendStatus === 'RECEIVE'}>
          {sendStatus === 'RECEIVE' ? '받은 쪽지' : '보낸 쪽지'}
        </Status>
        <Person>{sendStatus === 'RECEIVE' ? sender : '나'}</Person>
        <Time>{timeFormat(sendTime)}</Time>
      </Header>
      <Message>{message}</Message>
    </Note>
  );
}

function timeFormat(day: string) {
  let [date, time] = day.split('T');
  time = time.split(':').slice(0, 2).join(':');
  return [date, ' ', time];
}

const Note = styled.li`
  padding: 0.4rem 0;
  border-bottom: 1px solid black;
  color: ${({ theme }) => theme.textColor};
  font-size: ${({ theme }) => theme.fontRegular};
`;

const Header = styled.header`
  display: flex;
  align-items: flex-end;
  gap: 1rem;
`;

const Status = styled.span<{ status: boolean }>`
  font-size: ${({ theme }) => theme.fontSmall};
  color: ${({ status }) => (status ? '#42a5f5' : '#becf3d ')};
`;

const Person = styled.span`
  font-size: ${({ theme }) => theme.fontSmall};
`;

const Time = styled.span`
  font-size: ${({ theme }) => theme.fontSmall};
`;

const Message = styled.p`
  word-break: break-all;
`;
