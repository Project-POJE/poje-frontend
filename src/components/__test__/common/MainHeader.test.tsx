import '@testing-library/jest-dom';
import LoginView from '@src/components/common/LoginView';
import MainHeader from '@src/components/common/MainHeader';
import useUser from '@src/hooks/auth/useUser';
import { useNoteAlarm } from '@src/hooks/note';
import { withMemoryRouter } from '@src/test_util/util';
import { render, screen } from '@testing-library/react';
import { Route } from 'react-router-dom';

jest.mock('@src/hooks/auth/useUser', () => ({
  __esModule: true,
  default: jest.fn(),
}));
jest.mock('@src/hooks/note', () => ({
  __esModule: true,
  useNoteAlarm: jest.fn(),
}));

jest.mock('@src/components/common/LoginView', () => ({
  __esModule: true,
  default: jest.fn(),
}));

type MockedUseUser = () => { user: any };

// mock 메서드를 사용하도록 타입 추가
const mockedUseUser =
  useUser as MockedUseUser as jest.MockedFunction<MockedUseUser>;
const mockedUseNoteAlarm = useNoteAlarm as jest.MockedFunction<
  typeof useNoteAlarm
>;

describe('MainHeader test', () => {
  afterEach(() => {
    mockedUseUser.mockReset();
    mockedUseNoteAlarm.mockReset();
  });

  it('올바른 초기 렌더', () => {
    mockedUseUser.mockReturnValue({ user: true });
    mockedUseNoteAlarm.mockReturnValue(false);

    const { asFragment } = render(
      withMemoryRouter(
        <>
          <Route path='/' element={<MainHeader isHomePath={true} />} />
        </>,
        { initialEntries: ['/'] }
      )
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('로그인돼 있다면 노트 아이콘 표기', () => {
    mockedUseUser.mockReturnValue({ user: true });
    mockedUseNoteAlarm.mockReturnValue(false);

    render(
      withMemoryRouter(
        <>
          <Route path='/' element={<MainHeader isHomePath={true} />} />
        </>,
        { initialEntries: ['/'] }
      )
    );

    expect(screen.getByTitle('note-icon')).toBeInTheDocument();
  });

  it('로그인돼 있다면 노트 아이콘 표기 안함', () => {
    mockedUseUser.mockReturnValue({ user: false });
    mockedUseNoteAlarm.mockReturnValue(false);

    render(
      withMemoryRouter(
        <>
          <Route path='/' element={<MainHeader isHomePath={true} />} />
        </>,
        { initialEntries: ['/'] }
      )
    );

    //찾는 요소가 없으면 query로 했음 get은 요소가 없으면 에러를 발생하지만 query는 발생안함
    expect(screen.queryByTitle('note-icon')).not.toBeInTheDocument();
  });

  it('로그인돼 있고 새로운 쪽지가 있다면 N 표시', () => {
    mockedUseUser.mockReturnValue({ user: true });
    mockedUseNoteAlarm.mockReturnValue(true);

    render(
      withMemoryRouter(
        <>
          <Route path='/' element={<MainHeader isHomePath={true} />} />
        </>,
        { initialEntries: ['/'] }
      )
    );

    expect(screen.getByText('N')).toBeInTheDocument();
  });

  it('로그인돼 있고 새로운 쪽지가 없다면 N 표시 안함', () => {
    mockedUseUser.mockReturnValue({ user: false });
    mockedUseNoteAlarm.mockReturnValue(true);

    const { asFragment } = render(
      withMemoryRouter(
        <>
          <Route path='/' element={<MainHeader isHomePath={true} />} />
        </>,
        { initialEntries: ['/'] }
      )
    );

    expect(screen.queryByText('N')).not.toBeInTheDocument();
  });
});
