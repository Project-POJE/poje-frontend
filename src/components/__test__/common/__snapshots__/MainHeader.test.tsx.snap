// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainHeader test 올바른 초기 렌더 1`] = `
<DocumentFragment>
  <header
    class="sc-bcXHqe cFjoJE"
  >
    <div
      class="sc-gswNZR hTJXxB"
    >
      POJE
    </div>
    <div
      class="sc-dkrFOg iDgznT"
    >
      <nav
        class="sc-hLBbgP kaFhub"
      >
        <button
          class="sc-eDvSVe czqesj"
          data-type="note"
        >
          <svg
            class="icon"
            data-type="note"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              note-icon
            </title>
            <rect
              fill="none"
              height="320"
              rx="40"
              ry="40"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="32"
              width="416"
              x="48"
              y="96"
            />
            <path
              d="M112 160l144 112 144-112"
              fill="none"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="32"
            />
          </svg>
        </button>
      </nav>
    </div>
  </header>
</DocumentFragment>
`;
